File: app/src/main/java/com/popalay/cardme/utils/StringUtils.java
package com.popalay.cardme.utils;

import android.text.TextUtils;

import com.annimon.stream.Optional;
import com.annimon.stream.Stream;

public class StringUtils {

    private StringUtils() {}

    public static String getFirstLetters(String source) {
        final String[] split = TextUtils.split(source, " ");
        final Optional<String> single = Stream.of(split)
                .map(s -> s.charAt(0)).map(String::valueOf)
                .reduce((value1, value2) -> value1 + value2);
        return single.isPresent() ? single.get() : null;
    }
}

File: app/src/main/java/com/popalay/cardme/utils/StringUtils.kt
package com.popalay.cardme.utils

fun String.firstLetters(maxLetters: Int = 2): String {
    return if (this.isBlank()) "" else split(" ")
            .take(maxLetters)
            .map { it[0].toString() }
            .reduce { acc, s -> acc + s }
}

fun String.clean(): String {
    return this.trim().replace(Regex("\\s+"), " ")
}